gw_eccentricity/eccDefinition.py

```diff
@@ -451,6 +481,13 @@ def __init__(self, dataDict, num_orbits_to_exclude_before_merger=2,
         # compute residual data
         if "amplm_zeroecc" in self.dataDict and "omegalm_zeroecc" in self.dataDict:
             self.compute_res_amp_gw_and_res_omega_gw()
+        # use a flag so that the function
+        # `get_segment_of_data_for_finding_extrema` is called only once to get
+        # the relevant segment of data around the reference point.
+        # The function `get_segment_of_data_for_finding_extrema` is called from
+        # within the function `find_extrema`. This flag is set to False after
+        # `find_extrema` is called for the first time.
+        self.get_segment_of_data = self.extra_kwargs["use_only_these_many_orbits"]
```

Comment:
Why not call this self.use_only_these_many_orbits? They are supposed to mean the same thing.